File: hw/aimacana/u3/hw25CompositePattern/python/main.py
import sys
import os
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from model.teller import Teller
from model.clerk import Clerk
from model.manager import Manager
from model.president import President
from view.organization_view import OrganizationView
from controller.organization_controller import OrganizationController

def main():
    lonny = Teller("Lonny")
    cal = Clerk("Cal")
    able = Manager("Able")
    able.add(lonny)
    able.add(cal)
    
    juanita = Teller("Juanita")
    tina = Teller("Tina")
    thelma = Teller("Thelma")
    becky = Manager("Becky")
    becky.add(juanita)
    becky.add(tina)
    becky.add(thelma)
    
    pete = President.get_president("Pete")
    pete.add(able)
    pete.add(becky)
    
    view = OrganizationView()
    controller = OrganizationController(pete, view)
    
    controller.update_view()

if __name__ == "__main__":
    main()

File: hw/aimacana/u3/hw25CompositePattern/javascript/main.js
const Teller = require('./model/Teller');
const Clerk = require('./model/Clerk');
const Manager = require('./model/Manager');
const President = require('./model/President');
const OrganizationView = require('./view/OrganizationView');
const OrganizationController = require('./controller/OrganizationController');

function main() {
    const lonny = new Teller("Lonny");
    const cal = new Clerk("Cal");
    const able = new Manager("Able");
    able.add(lonny);
    able.add(cal);
    
    const juanita = new Teller("Juanita");
    const tina = new Teller("Tina");
    const thelma = new Teller("Thelma");
    const becky = new Manager("Becky");
    becky.add(juanita);
    becky.add(tina);
    becky.add(thelma);
    
    const pete = President.getPresident("Pete");
    pete.add(able);
    pete.add(becky);
    
    const view = new OrganizationView();
    const controller = new OrganizationController(pete, view);
    
    controller.updateView();
}

main();
